src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
 
-}
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.isInvalid()) {
```

Comment:
usually it is done this way. It is much better to start booleans with positive - as it is much easier to process in human mind
```suggestion
    if (!this.isValid()) {
```

Reply:
@naziamoff Always break a leg in such things